[WebDriver] Update imported WPT WebDriver tests - W47 2025

lauromoura opened the pull request:
#### 5eb76deba550516e5a0c5210392916dde115fb8b
<pre>
[WebDriver] Update imported WPT WebDriver tests - W47 2025
<a href="https://bugs.webkit.org/show_bug.cgi?id=301583">https://bugs.webkit.org/show_bug.cgi?id=301583</a>

Reviewed by BJ Burg.

Updated to commit 3d8fdcd7ec, from Nov 19th, 2025.

Some relevant changes since last update in May:

- New tests for a number of new features. Most of these new features are
  skipped, as they&apos;re not implemented yet.
  - bluetooth
  - emulation.set_locale_override
  - emulation.set_scripting_enabled
  - emulation.set_screen_orientation_override
  - emulation.set_timezone_override
  - emulation.set_useragent_override
  - emulation.set_network_conditions
  - network.add_data_collector
  - network.remove_data_collector
  - network.disown_data
  - network.get_data
  - network.set_extra_headers
- New tests for context created subscription steps.
- BrowsingContext.Info assertions now unconditionally check the
  clientWindow. Opened webkit.org/b/303210 to handle a few regressions that
  appeared.
- Unsubscription now is either by subscription ID or the event name, so
  the tests for unsubscribing by context were removed.

* Tools/Scripts/webkitpy/webdriver_tests/webdriver_w3c_executor.py:
(WebKitDriverBrowser.__init__): Add placeholder for required parameter
manager_number. As we run a single worker, this should not be an issue
for now.
* WebDriverTests/TestExpectations.json: Gardened the BiDi results.
* WebDriverTests/imported/w3c/: Update imported code.

Canonical link: <a href="https://commits.webkit.org/303718@main">https://commits.webkit.org/303718@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f1cf6327ebce93e51bb1945558bb117494745876

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133364 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5865 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44500 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140919 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/85411 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/135234 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/6378 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5730 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/102020 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/85411 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136311 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/4532 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119543 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82811 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/132713 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/4411 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1995 "Passed tests") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/113507 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/37651 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143566 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5535 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/38237 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110393 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5617 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4752 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110572 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28032 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4275 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115798 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/59285 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5590 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/34135 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5436 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5679 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5546 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

burg reviewed (approved):
I haven't looked too deeply into the changes, but I'm excited to get more recent WebDriver BiDI tests in the tree.

lauromoura committed:
[WebDriver] Update imported WPT WebDriver tests - W47 2025
https://bugs.webkit.org/show_bug.cgi?id=301583

Reviewed by BJ Burg.

Updated to commit 3d8fdcd7ec, from Nov 19th, 2025.

Some relevant changes since last update in May:

- New tests for a number of new features. Most of these new features are
  skipped, as they're not implemented yet.
  - bluetooth
  - emulation.set_locale_override
  - emulation.set_scripting_enabled
  - emulation.set_screen_orientation_override
  - emulation.set_timezone_override
  - emulation.set_useragent_override
  - emulation.set_network_conditions
  - network.add_data_collector
  - network.remove_data_collector
  - network.disown_data
  - network.get_data
  - network.set_extra_headers
- New tests for context created subscription steps.
- BrowsingContext.Info assertions now unconditionally check the
  clientWindow. Opened webkit.org/b/303210 to handle a few regressions that
  appeared.
- Unsubscription now is either by subscription ID or the event name, so
  the tests for unsubscribing by context were removed.

* Tools/Scripts/webkitpy/webdriver_tests/webdriver_w3c_executor.py:
(WebKitDriverBrowser.__init__): Add placeholder for required parameter
manager_number. As we run a single worker, this should not be an issue
for now.
* WebDriverTests/TestExpectations.json: Gardened the BiDi results.
* WebDriverTests/imported/w3c/: Update imported code.

Canonical link: https://commits.webkit.org/303718@main
